tfexec/show.go

```diff
@@ -10,6 +10,8 @@ import (
 	tfjson "github.com/hashicorp/terraform-json"
 )
 
+// Show reads the default state path and outputs the state.
+// To read a state or plan file, ShowState or ShowPlan must be used instead.
 func (tf *Terraform) Show(ctx context.Context) (*tfjson.State, error) {
```

Comment:
Should we just remove this version?

Reply:
No, `terraform show -json` is a valid use case. If `Show()` were removed, consumers would have to call either:
 - `ShowState(ctx, "terraform.tfstate")`, exposing the name of the state file, which is an implementation detail, or 
 - `ShowState(ctx, "")`, which users would expect to call `terraform show -json ""`, but would in fact be a special case calling `terraform show -json`, since the former leads to a CLI error.

Reply:
Makes sense, I wonder if the other two methods should be called `ShowPlanFile`, `ShowStateFile`, i just worry it could be confusing as to what they are calling under the hood and why they are differentiated.

Reply:
Should you error out if they pass a blank?